Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -366,6 +365,37 @@ - (void)setImageAnalysis:(CocoaImageAnalysis *)analysis;
 
 #if ENABLE(IMAGE_ANALYSIS)
 
+class ImageAnalysisCallbackAggregator : public ThreadSafeRefCounted<ImageAnalysisCallbackAggregator> {
```

Comment:
You could just use:
ThreadSafeRefCounted<ImageAnalysisCallbackAggregator, WTF::DestructionThread::MainRunLoop>

Then the destructor only needs to call the completion handler, no need for threading checks.

Reply:
Maybe we could even templatize the WTF one to promote reuse:
```
diff --git a/Source/WTF/wtf/CallbackAggregator.h b/Source/WTF/wtf/CallbackAggregator.h
index 88c5bbd093ba..cca14ecb5271 100644
--- a/Source/WTF/wtf/CallbackAggregator.h
+++ b/Source/WTF/wtf/CallbackAggregator.h
@@ -32,7 +32,8 @@
 
 namespace WTF {
 
-class CallbackAggregator : public ThreadSafeRefCounted<CallbackAggregator> {
+template <DestructionThread destructionThread>
+class CallbackAggregatorOnThread : public ThreadSafeRefCounted<CallbackAggregator, destructionThread> {
 public:
     static Ref<CallbackAggregator> create(CompletionHandler<void()>&& callback) { return adoptRef(*new CallbackAggregator(WTFMove(callback))); }
 
@@ -44,7 +45,7 @@ public:
     }
 
 private:
-    explicit CallbackAggregator(CompletionHandler<void()>&& callback)
+    explicit CallbackAggregatorOnThread(CompletionHandler<void()>&& callback)
         : m_callback(WTFMove(callback))
 #if ASSERT_ENABLED
         , m_wasConstructedOnMainThread(isMainThread())
@@ -58,6 +59,9 @@ private:
 #endif
 };
 
+using CallbackAggregator = CallbackAggregatorOnThread<DestructionThread::Any>;
+using MainRunLoopCallbackAggregator = CallbackAggregatorOnThread<DestructionThread::MainRunLoop>;
+
 } // namespace WTF
 
 using WTF::CallbackAggregator;
```

Reply:
Yeah, I considered adding it to `WTF::CallbackAggregator`, but wasn't sure if I should since we're only using it in this file. Sounds like this is okay though, so I'll update this patch to use `MainRunLoopCallbackAggregator`. Thanks!

Reply:
I think this is useful and likely to be reusable. As a matter of fact, I think `RemovalCallbackAggregator` in WebsiteDataStore.cpp might be able to use it.

Reply:
Hm...that helper class already seems to assert that the destructor is only called on the main runloop, though:
```
    ~RemovalCallbackAggregator()
    {
        ASSERT(RunLoop::isMain());
        RunLoop::main().dispatch(WTFMove(m_completionHandler));
    }
```
(Though, maybe there are still cases where it can be destroyed on the background, perhaps via API misuse by the client?)

Reply:
Because it is asking for its destructor to be called on the main run loop via its base class:
`class RemovalCallbackAggregator : public ThreadSafeRefCounted<RemovalCallbackAggregator, WTF::DestructionThread::MainRunLoop> {`

Reply:
The difference is that `~MainRunLoopCallbackAggregator()` may be called on the background though, whereas  `~RemovalCallbackAggregator()` is intended to be invoked on the main runloop only (the thing they do have in common is that both will invoke their callbacks on the main runloop).

I think I would prefer to replace `RemovalCallbackAggregator` with `MainRunLoopCallbackAggregator` in a separate patch, since it's a very subtle behavior change (and isn't necessary to fix this particular crash).